File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.js
class Solution {
    /**
     * @param {number[]} prices
     * @return {number}
     */
    maxProfit(prices) {
        const length = prices.length;
        let profit = 0;
        if(length == 1){
            return profit;
        }
        let buyValue = prices[0];
        for(let i=1; i<length; i++){
            buyValue = Math.min(buyValue,prices[i]);
            profit = Math.max(profit,prices[i]-buyValue);
        }
        return profit;
    }
}

File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-2.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int length = cost.length;
        if(length == 2){
            return Math.min(cost[0],cost[1]);
        }

        int[]dp = new int[length];
        dp[length-1] = cost[length-1];
        dp[length-2] = cost[length-2];

        for(int i=length-3; i>=0; i--){
            dp[i] = Math.min(cost[i]+dp[i+1],cost[i]+dp[i+2]);
        }
        return Math.min(dp[0],dp[1]); 
    }
}
